src/overlays/actors/ovl_En_Test/z_en_test.c

```diff
@@ -133,126 +204,1885 @@ static ColliderQuadInit D_808645C8 = {
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_JUMP);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    EnTest_SetupAction(this, EnTest_JumpUp);
+}
+
+void EnTest_JumpUp(EnTest* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 1, 0xFA0, 1);
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->actor.world.pos.y <= this->actor.floorHeight) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_DODO_M_GND);
+        this->actor.shape.rot.y = this->actor.yawTowardsPlayer;
+        this->actor.world.pos.y = this->actor.floorHeight;
+        this->unk_7E4 = -(s32)this->actor.velocity.y;
+
+        if (this->unk_7E4 == 0) {
+            this->unk_7E4 = 1;
+        }
+
+        this->actor.velocity.y = 0.0f;
+        this->actor.speedXZ = 0.0f;
+        this->unk_7C8 = 0xC;
+        this->timer = 4;
+        Animation_Change(&this->skelAnime, &gStalfosLandFromLeapAnim, 0.0f, 0.0f, 0.0f, 2, 0.0f);
+        EnTest_SetupAction(this, EnTest_Land);
+    }
+}
+
+void EnTest_SetupStopAndBlock(EnTest* this) {
+    Animation_Change(&this->skelAnime, &gStalfosBlockWithShieldAnim, 2.0f, 0.0f,
+                     Animation_GetLastFrame(&gStalfosBlockWithShieldAnim), 2, 2.0f);
+    this->unk_7C8 = 0x15;
+    this->actor.speedXZ = 0.0f;
+    this->timer = (Rand_ZeroOne() * 10.0f) + 11.0f;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->unk_7DE = 5;
+    EnTest_SetupAction(this, EnTest_StopAndBlock);
+}
+
+void EnTest_StopAndBlock(EnTest* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 1.0f, 0.5f, 0.0f);
+    SkelAnime_Update(&this->skelAnime);
+
+    if ((ABS((s16)(this->actor.yawTowardsPlayer - this->actor.shape.rot.y)) > 0x3E80) &&
+        (this->actor.params != STALFOS_TYPE_CEILING) && ((globalCtx->gameplayFrames % 2) != 0)) {
+        this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+        EnTest_SetupJumpBack(this);
+    }
+
+    if (this->timer == 0) {
+        EnTest_SetupIdleFromBlock(this);
+    } else {
+        this->timer--;
+    }
+}
+
+void EnTest_SetupIdleFromBlock(EnTest* this) {
+    Animation_MorphToLoop(&this->skelAnime, &gStalfosMiddleGuardAnim, -4.0f);
+    this->unk_7C8 = 0x16;
+    EnTest_SetupAction(this, EnTest_IdleFromBlock);
+}
+
+void EnTest_IdleFromBlock(EnTest* this, GlobalContext* globalCtx) {
+    Math_SmoothStepToF(&this->actor.speedXZ, 0.0f, 1.0f, 1.5f, 0.0f);
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->skelAnime.morphWeight == 0.0f) {
+        this->actor.speedXZ = 0.0f;
+        this->unk_7DE = 0;
+
+        if (!EnTest_ReactToProjectile(globalCtx, this)) {
+            if (this->actor.xzDistToPlayer < 500.0f) {
+                EnTest_ChooseAction(this, globalCtx);
+            } else {
+                EnTest_SetupSidestepAgro(this, globalCtx);
+            }
+        }
+    }
+}
+
+// setup damaged
+void func_80862154(EnTest* this) {
```

Comment:
Any reason for not just naming this?
Same for `func_808621D4`

Reply:
the comments have since been removed. there are 3 different functions that handle different kinds of damage, so they need to be figured out